File: sobit_mini_library/example/test_control_wheel.py
#!/usr/bin/env python3
import rospy
from sobit_mini_module import SobitMiniWheelController
import sys

def test():
    rospy.init_node('test')
    args = sys.argv
    mini_wheel_ctr = SobitMiniWheelController(args[0]) # args[0] : C++上でros::init()を行うための引数
    
    # タイヤ車輪をを動かす
    mini_wheel_ctr.controlWheelLinear(1.0)
    mini_wheel_ctr.controlWheelRotateRad(1.57)
    mini_wheel_ctr.controlWheelRotateDeg(-90)

    mini_wheel_ctr.controlWheelLinear(-1.0)

if __name__ == '__main__':
    try:
        test()
    except rospy.ROSInterruptException: pass

File: sobit_mini_library/example/test_control_wheel.cpp
#include "rclcpp/rclcpp.hpp"
#include "sobit_mini_library/sobit_mini_wheel_controller.hpp"

int main(int argc, char *argv[]) {
  // ROS2の初期化
  rclcpp::init(argc, argv);

  // ノードの作成
  auto node = std::make_shared<sobit_mini::WheelController>();

  // 車輪制御インスタンスの作成
  // `WheelController`クラスが提供するメソッドを利用します

  // 直線移動 (linear motion)
  node->controlWheelLinear(0.5);

  // 回転移動 (radian) (rotational motion: Radian)
  node->controlWheelRotateRad(-1.5708);

  // 回転移動 (degree) (rotational motion: Degree)
  node->controlWheelRotateDeg(90);

  // 直線移動 (linear motion)
  node->controlWheelLinear(-0.5);

  // ノードの終了処理 (Node termination)
  node.reset();

  // ROS2の終了処理 (ROS2 termination)
  rclcpp::shutdown();

  return 0;
}
